src/lib/utils/resource-loader.ts

```diff
@@ -11,81 +11,148 @@
 
 import * as path from 'path';
 
-import * as _ from 'lodash';
 import * as globby from 'globby';
 
+import { findPackageRoot } from './misc';
 import { debug as d } from './debug';
 import { ICollectorBuilder, IFormatter, IPluginBuilder, Resource, IRuleBuilder } from '../types'; // eslint-disable-line no-unused-vars
+import { validate as validateRule } from '../config/config-rules';
+
 
 const debug = d(__filename);
+const PROJECT_ROOT = findPackageRoot();
+
+const resources = new Map();
 
 /** The type of resource */
-const TYPE = {
+export const TYPE = {
     collector: 'collector',
     formatter: 'formatter',
-    plugin: 'plugin',
     rule: 'rule'
 };
 
-/** Loads all the resources available for the given type */
-const loadOfType = (type: string): Map<string, Resource> => {
-    const PROJECT_ROOT = path.join(__dirname, '../../../../');
+const getCoreResources = (type: string): Array<string> => {
+    const resourcesFiles: string[] = globby.sync(`${PROJECT_ROOT}/dist/src/lib/${type}s/**/*.js`, { absolute: true });
```

Comment:
I think it would be nice to have some kind of cache here instead of calculate every time the core resources